### PR TITLE
Task 1: Unit tests for common functions

### PR DESCRIPTION
**Pull request by Rahul-JOON**
This PR contains the test cases for the common functions as mentioned in issue #99.

Because the codebase is not broken, all of the test cases are now passing.

### CODE REVIEW

File: tests/test_common_functions.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    _dot_product,
+    _norm,
+    _clip,
+    _out_of_bounds,
+    _find_min_dist,
+    _aabbs_not_intersecting,
+)
+
+
+class Testdotproduct:
```
Comment (armantekinalp) [inline tests/test_common_functions.py:16]:
```suggestion
class TestDotProduct:
```

File: tests/test_common_functions.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        assert_allclose(dot_product, -1)
+
+        "test for complex vectors"
+        vec1 = [1, -2, 3]
+        vec2 = [-2, 1, 3]
+        dot_product = _dot_product(vec1, vec2)
+        assert_allclose(dot_product, 5)
+
+
+class Testnorm:
```
Comment (armantekinalp) [inline tests/test_common_functions.py:60]:
```suggestion
class TestNorm:
```

File: tests/test_common_functions.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    )
+    def test_out_of_bounds_with_verified_values(self, x, result):
+        "testing function with analytically verified values"
+
+        low = 1.0
+        high = 2.0
+        assert _out_of_bounds(x, low, high) == result
+
+
+class Testfinmindist:
```
Comment (armantekinalp) [inline tests/test_common_functions.py:117]:
```suggestion
class TestFindMinDist:
```

**Review by armantekinalp — CHANGES_REQUESTED** [review]
There are couple of minor changes you can see the suggestions.

Also can you rename file name to `test_contact_common_functions.py`.

**Review by armantekinalp — APPROVED** [review]
LGTM

**Review by bhosale2 — APPROVED** [review]
LGTM. Good job @Rahul-JOON. @armantekinalp you can go ahead with the merge.

**Pull request merged**